src/lib/UI/LabelMaker/LabelMaker.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUi\UI\LabelMaker;
+
+/**
+ * Makes human readable labels for UI items given an item and a label identifier.
+ */
+interface LabelMaker
```

Comment:
Not sure how I feel about `Maker`/`make` naming convention. Have you considered any other, like ie. `Generator`, `Provider`, `Formatter` (and such)?

Edit: You do use `generate` verb in ie. `bool $generateDefault = true` (an argument of interface method) and docblock. I would replace `Maker`/`make` with `Generator`/`generate`.

Reply:
Well, "maker" was a reference to physical label makers. References are good, they make sense out of things. But I agree about the inconsistency with having a "maker" that "generates" stuff.